workflow-event-listener/src/main/java/com/netflix/conductor/contribs/listener/archive/ArchivingWorkflowListenerProperties.java

```diff
@@ -21,9 +21,19 @@
 
 @ConfigurationProperties("conductor.workflow-status-listener.archival")
 public class ArchivingWorkflowListenerProperties {
-
+    public enum ArchivalType {
+        DEFAULT,
+        S3
+    }
     private final Environment environment;
 
+    private ArchivalType workflowArchivalType = ArchivalType.DEFAULT;
+
+    private String workflowS3ArchivalDefaultBucketName = "";
+
+    private String workflowS3ArchivalBucketRegion = "us-east-1";
+
+    private Boolean workflowArchiveUnsuccessfulOnly = false;
```

Comment:
Why not pull the system property here?

Reply:
Are you saying that the setter methods will be called when this class is initialised and all the values will get set?

Can they be set by the constructor and not by the setter methods?

Reply:
Yes the setter methods are called when the server start up and it finds the appropriate properties in config file. Also if you see the variable names are the exact same as the property defined in config file for example, workflowS3ArchivalBucketRegion, etc. This is how springboot framework binds the variables with property values. So I don't think they can be setup in constructor.

Reply:
Okay. Sounds good.